File: Assets/common/script/PersistentScene.js
#pragma strict
static var networkEnabled : boolean = true;
static var networkReady : boolean = false;
/* global things that persist between scenese
*/
var hyperspaceDestination : int; //destination for our hyperspace exit
var forcedHyperspaceFail : boolean; //do we force a failure?
var useNetwork : boolean = true;
public var deathReason : String = "";
public var survivedTheGame : boolean = false;

public var pilotName : String = "pilotname";
public var tacticalName : String = "tacticalname";
public var engineerName : String = "engineername";
public var captainName : String = "captainname";
public var gmName : String = "gmname";


public static var _instance : PersistentScene;



@HideInInspector




function Awake(){
	DontDestroyOnLoad(this);
	networkEnabled = useNetwork;
	_instance = this;
}

function Start () {
	if(useNetwork == false){
		networkReady = false;
	}

}

function Update () {

}
function gameWin(){
	survivedTheGame = true;
	Application.LoadLevel(5);
}

/* we died, do the global OHYOUBEDEAD things */
function shipDead(reasonText : String){
	var msg : OSCMessage = OSCMessage("/scene/youaredead");
	msg.Append.<String>(reasonText);
	OSCHandler.Instance.SendMessageToAll(msg);
	deathReason = reasonText;
	Application.LoadLevel(5);

}

File: Assets/common/script/PersistentScene.cs
using UnityEngine;
using System;
using UnityOSC;


public class PersistentScene:MonoBehaviour{
	public static bool networkEnabled = true;
	public static bool networkReady = false;
	/* global things that persist between scenese
	*/
	//TODO change this to a string value naming the scene
	public string hyperspaceDestination; //destination for our hyperspace exit
	public bool forcedHyperspaceFail; //do we force a failure?
	public bool useNetwork = true;
	public string deathReason = "";
	public bool survivedTheGame = false;
	
	public string pilotName = "pilotname";
	public string tacticalName = "tacticalname";
	public string engineerName = "engineername";
	public string captainName = "captainname";
	public string gmName = "gmname";
	
	
	public static PersistentScene _instance;
	
	
	
	[HideInInspector]
	
	
	
	
	public void Awake(){
		DontDestroyOnLoad(this);
		networkEnabled = useNetwork;
		_instance = this;
	}
	
	public void Start() {
		if(useNetwork == false){
			networkReady = false;
		}
		Cursor.visible = false;
	
	}
	
	public void Update() {
	
	}
	public void gameWin(){
		survivedTheGame = true;
		Application.LoadLevel("deadscene");
	}
	
	/* we died, do the global OHYOUBEDEAD things */
	public void shipDead(string reasonText){
		OSCMessage msg = new OSCMessage("/scene/youaredead");
		msg.Append<String>(reasonText);
		OSCHandler.Instance.SendMessageToAll(msg);
		deathReason = reasonText;
		Application.LoadLevel("deadscene");
	
	}
}